nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -2,29 +2,85 @@
 
 import actionlib
 import rospy
+import math
+import time
 
-# joint
+# Joint
 from moveit_action_handlers.msg import PropertyValuePair
 from moveit_action_handlers.msg import MoveToJointsMoveItAction
 from moveit_action_handlers.msg import MoveToJointsMoveItGoal
 
-from sensor_msgs.msg import JointState
-
-# cartesian
+# Cartesian
 from moveit_action_handlers.msg import MoveToPoseMoveItAction
 from moveit_action_handlers.msg import MoveToPoseMoveItGoal
 from moveit_action_handlers.msg import PoseStamped
-from geometry_msgs.msg import Pose
-
-# forward kinematics
-# from urdf_parser_py.urdf import URDF
-# from pykdl_utils.kdl_kinematics import KDLKinematics
 
+# Kinematics
+from sensor_msgs.msg import JointState
+import tf_conversions, tf2_ros
+from geometry_msgs.msg import Pose as GeometryPose
+from geometry_msgs.msg import Quaternion, Point
+from tf.transformations import euler_from_quaternion
 
 # TODO: replace with service call, requires refactoring of service to include gripper info
+CFG = rospy.get_param("/nightingale_utils/joint_configurations")
```

Comment:
- [ ] this shoud be deleted and use the service instead

Reply:
or maybe the caller should use the for loop

Reply:
done in lyndons pr